lib/ansible/modules/cloud/cloudstack/cs_vlan_ip_range.py

```diff
@@ -0,0 +1,327 @@
+  returned: success
+  type: str
+  sample: vlan://98
+gateway:
+  description: IPv4 gateway.
+  returned: success
+  type: str
+  sample: 10.2.4.1
+netmask:
+  description: IPv4 netmask.
+  returned: success
+  type: str
+  sample: 255.255.255.0
+gateway_ipv6:
+  description: IPv6 gateway.
+  returned: success
+  type: str
+  sample: 2001:db8::1
+zone:
+  description: Name of zone.
+  returned: success
+  type: str
+  sample: zone-02
+domain:
+  description: Domain name of the VLAN IP range.
+  returned: success
+  type: str
+  sample: ROOT
+account:
+  description: Account who owns the network.
+  returned: success
+  type: str
+  sample: example account
+project:
+  description: Project who owns the network.
+  returned: success
+  type: str
+  sample: example project
+for_systemvms:
+  description: Whether VLAN IP range is dedicated to system vms or not.
+  returned: success
+  type: bool
+  sample: false
+for_virtual_network:
+  description: Whether VLAN IP range is of Virtual type or not.
+  returned: success
+  type: bool
+  sample: false
+physical_network:
+  description:  The physical network VLAN IP range belongs to.
+  returned: success
+  type: str
+  sample: 04589590-ac63-4ffc-93f5-b698b8ac38b6
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.cloudstack import (
+    AnsibleCloudStack,
+    cs_argument_spec,
+    cs_required_together,
+)
+
+
+class AnsibleCloudStackVlanIpRange(AnsibleCloudStack):
+
+    def __init__(self, module):
+        super(AnsibleCloudStackVlanIpRange, self).__init__(module)
+        self.returns = {
+            'startip': 'start_ip',
+            'endip': 'end_ip',
+            'physicalnetworkid': 'physical_network',
+            'vlan': 'vlan',
+            'forsystemvms': 'for_systemvms',
+            'forvirtualnetwork': 'for_virtual_network',
+            'gateway': 'gateway',
+            'netmask': 'netmask',
+            'ip6gateway': 'gateway_ipv6',
+            'startipv6': 'start_ipv6',
+            'endipv6': 'end_ipv6',
+        }
+        self.ip_range = None
+
+    def get_vlan_ip_range(self):
+        if not self.ip_range:
+            args = {
+                'zoneid': self.get_zone(key='id'),
+                'projectid': self.get_project(key='id'),
+                'account': self.get_account(key='name'),
+                'domainid': self.get_domain(key='id'),
+                'networkid': self.get_network(key='id'),
+            }
+
+            res = self.query_api('listVlanIpRanges', **args)
+            if res:
+                ip_range_list = res['vlaniprange']
+
+                params = {
+                    'startip': self.module.params.get('start_ip'),
+                    'endip': self.get_or_fallback('end_ip', 'start_ip'),
+                }
+
+                for ipr in ip_range_list:
+                    if params['startip'] == ipr['startip'] and params['endip'] == ipr['endip']:
+                        self.ip_range = ipr
```

Comment:
break the loop in case we found the first match?

~~~diff
                     if params['startip'] == ipr['startip'] and params['endip'] == ipr['endip']:
                         self.ip_range = ipr
+                        break
~~~

Reply:
Indeed, no need to continue if we have a first match